### PR TITLE
Add and use stringize function

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Turns a bool into "true" or "false" string

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Utilities/GetOutput.hpp [hunk 1]
```diff
@@ -5,14 +5,19 @@
...
  * \ingroup UtilitiesGroup
  * \brief Get the streamed output of `t` as a `std::string`
  */
 template <typename T>
 std::string get_output(const T& t) {
-  std::ostringstream os;
-  os << t;
-  return os.str();
+  if constexpr (std::is_same_v<T, bool>) {
+    return t ? "true" : "false";
```
Comment (wthrowe) [inline src/Utilities/GetOutput.hpp]: The thing you're looking for is called `std::boolalpha`.

**Review by wthrowe — CHANGES_REQUESTED** [review]
I don't think this is a good idea.  `get_output` is used in many places for testing that objects print correctly, or for finding the result of printing the object for some other reason.  Making it return something other than the plain result of streaming breaks that.

File: src/Utilities/GetOutput.hpp (hunk 1 above)
Comment (knelli2) [reply src/Utilities/GetOutput.hpp]: Should I use `std::boolalpha` here then in the stream part, or just make a new `stringize` function in Utilities?

**Comment by nilsdeppe** [comment]
I agree with @wthrowe This could lead to confusion. I'm not sure what the right answer is because I also agree that being able to control the output would be nice. Possible `get_output` should take a stream modifier object? I haven't looked into that, so it might just not be possible

**Comment by wthrowe** [comment]
Sorry for the confusion.  The two parts of my comment were unrelated.  I don't think we should do this, but if we do, or add something similar, we should use the standard library functionality.

As to what to do, I think `(MakeString{} << std::boolalpha << flag)` would give the right result, and is probably simple enough to not need a wrapper function.

**Comment by knelli2** [comment]
Ok @wthrowe I left `get_output` alone and just added a global `stringize` function. I think it's worth having this because now all places that need to convert a bool to a string use the standard library functionality. Whereas previously none of them did

**Comment by wthrowe** [comment]
I think "stringize" is a really generic name for something that only works on bool.  I suggest one of (a) rename it to make it's purpose clear, (b) allow it to take any type so it works on `vector<bool>` and similar (although then people will probably confuse it with `get_output`), or (c) inline it everywhere.

I think this is mostly used for creating option strings for tests, so a function to convert simple types to something the YAML parser will understand might be a general solution.  The yaml-cpp library probably has something to do that, but if we only care about a few things like bools and vectors writing our own is probably easier.

**Comment by nilsvu** [comment]
Yea generating YAML output from types would be the way to go I think

**Comment by kidder** [comment]
I agree with this, I was going to suggest something like `Options::to_yaml`

**Comment by wthrowe** [comment]
As long as we don't want to do anything complicated so we care about indenting and stuff it looks like yaml-cpp makes that pretty easy: https://github.com/jbeder/yaml-cpp/wiki/How-To-Emit-YAML

**Pull request closed without merging**